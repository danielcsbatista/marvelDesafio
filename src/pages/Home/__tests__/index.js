import React from 'react';
import { Home } from '../index';
import categoryData from '../../../config/data/category';

describe('Home', () => {
  it('renders <Home /> without crashing', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  it('on click ListInitial call getCategory', () => {
    const getCategory = jest.fn();
    const navigate = jest.fn();

    const props = {
      categoryActions: { getCategory },
      category: { data: [] },
      navigation: { navigate },
    };

    const wrapper = shallow(<Home {...props} />);
    wrapper
      .find('ListInitial')
      .first()
      .props()
      .callFunction();
    const item = categoryData.data[0];
    const mock = {
      urlRefer: item.urlRefer,
      nameCategory: item.title,
      paramSearch: item.paramSearch,
      orderBy: item.orderBy,
    };
    expect(getCategory).toBeCalledWith(mock);
    expect(navigate).toBeCalledWith('ListContent');
  });
});
