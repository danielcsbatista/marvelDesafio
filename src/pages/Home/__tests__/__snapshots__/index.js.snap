// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders <Home /> without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Styled(ScrollView)>
        <ListInitial
          callFunction={[Function]}
          imgSrc={
            Object {
              "testUri": "../../../src/shared/images/characters.png",
            }
          }
          title="CHARACTERS"
        />
        <ListInitial
          callFunction={[Function]}
          imgSrc={
            Object {
              "testUri": "../../../src/shared/images/comics.png",
            }
          }
          title="COMICS"
        />
        <ListInitial
          callFunction={[Function]}
          imgSrc={
            Object {
              "testUri": "../../../src/shared/images/events.png",
            }
          }
          title="EVENTS"
        />
        <ListInitial
          callFunction={[Function]}
          imgSrc={
            Object {
              "testUri": "../../../src/shared/images/creators.png",
            }
          }
          title="CREATORS"
        />
        <ListInitial
          callFunction={[Function]}
          imgSrc={
            Object {
              "testUri": "../../../src/shared/images/series.png",
            }
          }
          title="SERIES"
        />
      </Styled(ScrollView)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/characters.png",
              }
            }
            title="CHARACTERS"
          />,
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/comics.png",
              }
            }
            title="COMICS"
          />,
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/events.png",
              }
            }
            title="EVENTS"
          />,
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/creators.png",
              }
            }
            title="CREATORS"
          />,
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/series.png",
              }
            }
            title="SERIES"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "callFunction": [Function],
            "imgSrc": Object {
              "testUri": "../../../src/shared/images/characters.png",
            },
            "title": "CHARACTERS",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "callFunction": [Function],
            "imgSrc": Object {
              "testUri": "../../../src/shared/images/comics.png",
            },
            "title": "COMICS",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "callFunction": [Function],
            "imgSrc": Object {
              "testUri": "../../../src/shared/images/events.png",
            },
            "title": "EVENTS",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "callFunction": [Function],
            "imgSrc": Object {
              "testUri": "../../../src/shared/images/creators.png",
            },
            "title": "CREATORS",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "callFunction": [Function],
            "imgSrc": Object {
              "testUri": "../../../src/shared/images/series.png",
            },
            "title": "SERIES",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Styled(ScrollView)>
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/characters.png",
              }
            }
            title="CHARACTERS"
          />
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/comics.png",
              }
            }
            title="COMICS"
          />
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/events.png",
              }
            }
            title="EVENTS"
          />
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/creators.png",
              }
            }
            title="CREATORS"
          />
          <ListInitial
            callFunction={[Function]}
            imgSrc={
              Object {
                "testUri": "../../../src/shared/images/series.png",
              }
            }
            title="SERIES"
          />
        </Styled(ScrollView)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ListInitial
              callFunction={[Function]}
              imgSrc={
                Object {
                  "testUri": "../../../src/shared/images/characters.png",
                }
              }
              title="CHARACTERS"
            />,
            <ListInitial
              callFunction={[Function]}
              imgSrc={
                Object {
                  "testUri": "../../../src/shared/images/comics.png",
                }
              }
              title="COMICS"
            />,
            <ListInitial
              callFunction={[Function]}
              imgSrc={
                Object {
                  "testUri": "../../../src/shared/images/events.png",
                }
              }
              title="EVENTS"
            />,
            <ListInitial
              callFunction={[Function]}
              imgSrc={
                Object {
                  "testUri": "../../../src/shared/images/creators.png",
                }
              }
              title="CREATORS"
            />,
            <ListInitial
              callFunction={[Function]}
              imgSrc={
                Object {
                  "testUri": "../../../src/shared/images/series.png",
                }
              }
              title="SERIES"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "callFunction": [Function],
              "imgSrc": Object {
                "testUri": "../../../src/shared/images/characters.png",
              },
              "title": "CHARACTERS",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "callFunction": [Function],
              "imgSrc": Object {
                "testUri": "../../../src/shared/images/comics.png",
              },
              "title": "COMICS",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "callFunction": [Function],
              "imgSrc": Object {
                "testUri": "../../../src/shared/images/events.png",
              },
              "title": "EVENTS",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "callFunction": [Function],
              "imgSrc": Object {
                "testUri": "../../../src/shared/images/creators.png",
              },
              "title": "CREATORS",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "callFunction": [Function],
              "imgSrc": Object {
                "testUri": "../../../src/shared/images/series.png",
              },
              "title": "SERIES",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [MockFunction],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
